src/routes/(authenticated)/profile/[user_id]/edit-profile/change-email/+page.server.ts

```diff
@@ -0,0 +1,72 @@
+import { zod } from 'sveltekit-superforms/adapters';
+import { message, superValidate } from 'sveltekit-superforms';
+import { fail, redirect, type Action } from '@sveltejs/kit';
+import { error } from '@sveltejs/kit';
+import type { PageServerLoad, Actions } from './$types';
+import { z } from 'zod';
+import { PUBLIC_BASE_URL } from '$env/static/public';
+
+const newEmail = z.object({
+	new_email: z.string().email({ message: 'Invalid email address' }),
+	confirm_email: z.string().email({ message: 'Invalid email address' })
+});
+
+export const load: PageServerLoad = async ({ cookies, locals }) => {
+	const user = locals.user;
+	
+	if (!user) {
+		return error(500, 'update e-mail: something went wrong on our side');
+	}
+	
+	const form = await superValidate(zod(newEmail));
+	console.log('HELLO FROM THE LOAD FUNCITON: form = ', form)
+	return { form };
+};
+
+export const actions: Actions = {
+	new_email: async ({
+		request,
+		cookies,
+		locals: { user, userRepository, emailRepository }
+	}) => {
+		const newemail = await superValidate(request, zod(newEmail));
+		
+		
+		if (!newemail) return fail(400, { newemail });
+
+		console.log('HELLO FROM THE FORM ACTION:', 'the formumlaire is valid')
+		if (newemail.data.new_email !== newemail.data.confirm_email) {
+			return message(newemail, 'new email and confirmation must be the same', {
+				status: 400
+			});
+		}
+
+		if (newemail.data.new_email == user.email) {
+			return message(newemail, 'new email must be different from the previous one', {
+				status: 400
+			});
+		}
+
+
+		const update =  userRepository.updateUserEmail(user.id, newemail.data.new_email);
+		if (!update.changes) {
+			console.log('IN THE CHANGE EMAIL ACTION FUNCTION update.change = ', update.changes)
+			error(500, 'something bad happens trying to update your email')
+		}
+
+
+		const res = emailRepository.updateEmailIsSetup(user.id, false);
+		if (!res.changes) {
+			console.log('IN THE CHANGE EMAIL ACTION FUNCTION EMAIL, res.changes = ', res.changes)
+			error(500, 'something bad happens trying to update your email status')
+		}
+
+		const verificationToken = emailRepository.createEmailVerificationToken(user.id, newemail.data.new_email);
```

Comment:
Some of these steps probably always go together. I would make them part of the email repository interface and do them together.
At the moment most of the email repo logic is in the UI. This should all live together in the repository and the ui should just call it and return the result to the user.

Also probably most of these things should be in a Service and the service should own a repository.
A repository should just store things, at the moment it does three things:
- send emails
- handle email resetting logic
- storing tokens etc